qdev_wrappers/tests/dataset/test_doNd.py

```diff
@@ -0,0 +1,92 @@
+"""
+These are the basic black box tests for the doNd functions.
+"""
+
+from qdev_wrappers.dataset.doNd import do0d, do1d, do2d
+from qcodes.instrument.parameter import Parameter
+from qcodes import config, new_experiment
+from qcodes.utils import validators
+
+import pytest
+
+config.user.mainfolder = "output"  # set ouput folder for doNd's
+new_experiment("doNd-tests", sample_name="no sample")
+
+_param = Parameter('simple_parameter',
+                   set_cmd=None,
+                   get_cmd=lambda: 1)
+
+_paramComplex = Parameter('simple_complex_parameter',
+                   set_cmd=None,
+                   get_cmd=lambda: 1 + 1j,
+                   vals=validators.ComplexNumbers())
+
+_param_set = Parameter('simple_setter_parameter',
+                       set_cmd=None,
+                       get_cmd=None)
+
+
+def _param_func():
+    """
+    A private utility function.
+    """
+    _new_param = Parameter('modified_parameter',
+                           set_cmd= None,
+                           get_cmd= lambda: _param.get()*2)
+    assert _new_param.get() == 2
+    return _new_param
+
+
+def test_do0d():
```

Comment:
You can shorten this test by parametrizing it.

Reply:
Does parametrization has a performance advantage? If not, I prefer to have every statement in the test case explicitly written.

Reply:
as per offline discussion: parameter in the sense of pytest.parametrize not qcodes.Parameter.